recipe/meta.yaml

```diff
@@ -35,51 +35,62 @@ outputs:
         - gitpython >=2.1.0,<4
-        - gunicorn <22  # [not win and mlflow_variant != "skinny"]
-        - numpy <2  # [mlflow_variant != "skinny"]
-        - pandas <3  # [mlflow_variant != "skinny"]
-        - prometheus_flask_exporter <1  # [mlflow_variant != "skinny"]
         - packaging <24
         - protobuf >=3.12.0,<5
         - python
         - pytz <2024
         - pyyaml >=5.1,<7
-        - querystring_parser <2  # [mlflow_variant != "skinny"]
-        - requests >=2.17.3,<3
-        - scipy <2  # [mlflow_variant != "skinny"]
-        - sqlalchemy >=1.4.0,<3  # [mlflow_variant != "skinny"]
         - sqlparse >=0.4.0,<1
-        - waitress <3  # [win and mlflow_variant != "skinny"]
+        - requests >=2.17.3,<3
         - importlib-metadata <7,>=3.7.0,!=4.7.0
         - openssl
-        - scikit-learn <2  # [mlflow_variant != "skinny"]
-        - pyarrow <13,>=4.0.0  # [mlflow_variant != "skinny"]
-        - markdown <4,>=3.3  # [mlflow_variant != "skinny"]
-        - jinja2 <4,>=3.0  # [win and mlflow_variant != "skinny"]
-        - jinja2 <4,>=2.11  # [not win and mlflow_variant != "skinny"]
-        - matplotlib-base <4  # [mlflow_variant != "skinny"]
+
+        - alembic <2,!=1.10             # [mlflow_variant != "skinny"]
+        - docker-py >=4.0.0,<7          # [mlflow_variant != "skinny"]
+        - flask <3                      # [mlflow_variant != "skinny"]
+        - gunicorn <22                  # [mlflow_variant != "skinny" and not win]
+        - jinja2 <4,>=3.0               # [mlflow_variant != "skinny" and win]
+        - jinja2 <4,>=2.11              # [mlflow_variant != "skinny" and not win]
+        - markdown <4,>=3.3             # [mlflow_variant != "skinny"]
+        - matplotlib-base <4            # [mlflow_variant != "skinny"]
+        - numpy <2                      # [mlflow_variant != "skinny"]
+        - pandas <3                     # [mlflow_variant != "skinny"]
+        - prometheus_flask_exporter <1  # [mlflow_variant != "skinny"]
+        - pyarrow <13,>=4.0.0           # [mlflow_variant != "skinny"]
+        - querystring_parser <2         # [mlflow_variant != "skinny"]
+        - scikit-learn <2               # [mlflow_variant != "skinny"]
+        - scipy <2                      # [mlflow_variant != "skinny"]
+        - sqlalchemy >=1.4.0,<3         # [mlflow_variant != "skinny"]
+        - waitress <3                   # [mlflow_variant != "skinny" and win]
+
+        # https://github.com/mlflow/mlflow/blob/master/requirements/gateway-requirements.txt
+        - pydantic >=1.0,<2             # [mlflow_variant != "skinny"]
+        - fastapi <1                    # [mlflow_variant != "skinny"]
+        - uvicorn <1                    # [mlflow_variant != "skinny"]
+        - watchfiles <1                 # [mlflow_variant != "skinny"]
+        - psutil <6                     # [mlflow_variant != "skinny"]
+        - aiohttp <4                    # [mlflow_variant != "skinny"]
```

Comment:
This should be covered by a separate `mlflow-gateway` output. You can take inspiration from the `black-feedstock`.

Reply:
That doesn't fix the problem that you can have `pydantic >=2` installed, which will break `import mlflow`. We'd have to remove `mlflow/gateway/` from the package. Shall I do that?

Reply:
In `black` we have a similar situation where all code is shipped in the main package but you will get an error if you only install `black` but call `blackd`. The situation with people using the `pip` package directly will be the same. The main difference between conda and pip is that in one case you would install `mlflow-gateway` and in the other `mlflow[gateway]`.